tpm2/tpm2.go

```diff
@@ -188,49 +188,62 @@ func ReadClock(rw io.ReadWriter) (uint64, uint64, error) {
 	return decodeReadClock(resp)
 }
 
-func decodeGetCapability(in []byte) (Capability, []tpmutil.Handle, error) {
+func decodeGetCapability(in []byte) (Capability, []interface{}, bool, error) {
 	var moreData byte
 	var capReported Capability
 
 	buf := bytes.NewBuffer(in)
 	if err := tpmutil.UnpackBuf(buf, &moreData, &capReported); err != nil {
-		return 0, nil, err
-	}
-	// Only TPM_CAP_HANDLES handled.
-	if capReported != CapabilityHandles {
-		return 0, nil, fmt.Errorf("Only TPM_CAP_HANDLES supported, got %v", capReported)
+		return capReported, nil, false, err
 	}
 
-	var numHandles uint32
-	if err := tpmutil.UnpackBuf(buf, &numHandles); err != nil {
-		return 0, nil, err
-	}
+	switch capReported {
+	case CapabilityHandles:
+		var numHandles uint32
+		if err := tpmutil.UnpackBuf(buf, &numHandles); err != nil {
+			return capReported, nil, false, fmt.Errorf("could not unpack handle count: %v", err)
+		}
 
-	var handles []tpmutil.Handle
-	for i := 0; i < int(numHandles); i++ {
-		var handle tpmutil.Handle
-		if err := tpmutil.UnpackBuf(buf, &handle); err != nil {
-			return 0, nil, err
+		var handles []interface{}
+		for i := 0; i < int(numHandles); i++ {
+			var handle tpmutil.Handle
+			if err := tpmutil.UnpackBuf(buf, &handle); err != nil {
+				return capReported, nil, false, fmt.Errorf("could not unpack handle: %v", err)
+			}
+			handles = append(handles, handle)
+		}
+		return capReported, handles, moreData > 0, nil
+	case CapabilityAlgs:
+		var numAlgs uint32
+		if err := tpmutil.UnpackBuf(buf, &numAlgs); err != nil {
+			return capReported, nil, false, fmt.Errorf("could not unpack algorithm count: %v", err)
 		}
-		handles = append(handles, handle)
-	}
 
-	return capReported, handles, nil
+		var algs []interface{}
+		for i := 0; i < int(numAlgs); i++ {
+			var alg AlgorithmDescription
+			if err := tpmutil.UnpackBuf(buf, &alg); err != nil {
+				return capReported, nil, false, fmt.Errorf("could not unpack algorithm description: %v", err)
+			}
+			algs = append(algs, alg)
+		}
+		return capReported, algs, moreData > 0, nil
+	default:
+		return capReported, nil, false, fmt.Errorf("unsupported capability %v", capReported)
+	}
 }
 
 // GetCapability returns various information about the TPM state.
```

Comment:
Mention supported capabilities and what types are expected in returned slice.
Also describe what moreData means and how to use it.